specs/eip4844/polynomial-commitments.md

```diff
@@ -117,9 +141,51 @@ def bit_reversal_permutation(sequence: Sequence[T]) -> Sequence[T]:
 ```python
 def bytes_to_bls_field(b: Bytes32) -> BLSFieldElement:
     """
-    Convert bytes to a BLS field scalar. The output is not uniform over the BLS field.
+    Convert 32-byte value to a BLS field scalar. The output is not uniform over the BLS field.
+    """
+    return int.from_bytes(b, ENDIANNESS) % BLS_MODULUS
+```
+
+#### `blob_to_field_elements`
+
+```python
+def blob_to_field_elements(blob: Blob) -> Polynomial:
+    """
+    Convert a blob to list of BLS field scalars.
+    """
+    r = Polynomial()
+    for i in range(FIELD_ELEMENTS_PER_BLOB):
+        value = int.from_bytes(blob[i * BYTES_PER_FIELD_ELEMENT: (i + 1) * BYTES_PER_FIELD_ELEMENT], ENDIANNESS)
+        assert value < BLS_MODULUS
+        r[i] = value
+    return r
+```
+
+#### `hash_to_bls_field`
+
+```python
+def hash_to_bls_field(polys: Sequence[Polynomial],
+                      comms: Sequence[KZGCommitment]) -> Tuple[BLSFieldElement, Hash32]:
+    """
+    Compute 32-byte hash of serialized polynomials and commitments concatenated.
+    This hash is then converted to a BLS field element, where the result is not uniform over the BLS field.
+    Return the BLS field element.
     """
-    return int.from_bytes(b, "little") % BLS_MODULUS
+    # Append the number of polynomials and the degree of each polynomial as a domain separator
```

Comment:
Optional: We could add an assert here `len(polys) == len(comms)`

Reply:
One reason to maybe not include it, is because this check is indirectly being done in `validate_blobs_and_kzg_commitments`

assert len(blob_kzg_commitments) == len(blobs)